back.py

```diff
@@ -78,3 +80,34 @@ def get_table_content(self, db_name, exp_names, columns):
         cursor = collection.find(filter=filter, projection=projection)
 
         return generate_experiment_table_rows(cursor, columns)
+
+    def filter_rows_by_query(self, rows, query):
+        '''
+        Returns rows that are filtered by the query.
+        Uses the pandas.DataFrame.query method
+        '''
+
+        if query.strip() == '':
+            return rows
+
+        df = DataFrame(rows)
+
+        # temporarily replacing '.' in columns to '__'
+        # for pandas syntax reasons
+        normalize_col_names = lambda x: x.replace('.', '__')
+        df = df.rename(columns=normalize_col_names)
+
+        # filter rows with query
+        try:
+            df_filtered = df.query(normalize_col_names(query), locals={})
```

Comment:
This line raises `eval() got an unexpected keyword argument 'locals'`
```suggestion
            df_filtered = df.query(normalize_col_names(query), local_dict={})
```